app/services/embedding_service.py

```
@@ -0,0 +1,138 @@
+            await session.commit()
+            await session.refresh(faq_embedding)
+
+            logger.info(f"FAQ embedding added with ID: {faq_embedding.id}")
+            return faq_embedding
+
+        except Exception as e:
+            logger.error(f"Error adding FAQ embedding: {e}")
+            await session.rollback()
+            return None
+
+    def _cosine_similarity(self, vec1: List[float], vec2: List[float]) -> float:
```

Ruff (0.12.2)

Use `list` instead of `List` for type annotation

Replace with `list`

(UP006)